core/src/main/java/org/pentaho/di/core/database/OracleDatabaseMeta.java

```diff
@@ -678,4 +678,8 @@ public boolean strictBigNumberInterpretation() {
   public void setStrictBigNumberInterpretation( boolean strictBigNumberInterpretation ) {
     getAttributes().setProperty( STRICT_BIGNUMBER_INTERPRETATION, strictBigNumberInterpretation ? "Y" : "N" );
   }
+
+  @Override public boolean supportsGetLengthByChar() {
+    return false;
```

Comment:
Same as above + this is where we need to read `ORACLEDB_LENGTH_SEMANTICS_USES_CHARS`. By default this method will return `false` but if the kettle property is defined it will return it's value.

This will then match the behavior of the Oracle DB property which is set to **bytes** by default. If for some reason the value has been changed to **chars**, then the user can set the kettle property to **Y**.